packages/patterns-react/src/patterns/TableOfContents/DesktopMenu.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+
+const { prefix } = settings;
+
+/**
+ * DesktopMenu Component
+ *
+ * @param {object} props props object
+ * @param {object} props.menuItems menu items object
+ * @param {string} props.selectedId id of a menu item
+ * @param {*} props.updateState function to update parent state.
+ * @returns {*} JSX Object
+ */
+const DesktopMenu = ({ menuItems, selectedId, updateState }) => {
+  /**
+   * Render menu items
+   *
+   * @param {Array} items menu items array
+   * @param {string} activeId selected item id
+   * @returns {*} JSX Object
+   */
+  const renderMenuItems = (items, activeId) => {
+    return items.map(item => {
+      if (item) {
+        return (
+          <li
+            key={item.id}
```

Comment:
@sannminwin let's also add a stable selector for the items as well.

Reply:
added stable selector

Reply:
added